wetterdienst/provider/dwd/derived/api.py

```diff
@@ -181,47 +384,54 @@ def _collect_station_parameter_or_dataset(
             date_range_to_fetch_for_period = self._filter_date_range_for_period(
                 date_range=full_date_range_to_fetch,
                 period=period,
-                parameter=parameter_or_dataset,
+                dataset=parameter_or_dataset.dataset,
             )
             for first_day_of_month_to_fetch in date_range_to_fetch_for_period:
-                for dataset, _ in groupby(self.sr.stations.parameters, key=lambda x: x.dataset):
-                    url = self._get_values_url(
-                        product_name=dataset.name,
-                        period=period,
-                        start_date=first_day_of_month_to_fetch.strftime("%Y%m"),
+                url = self._get_values_url(
+                    dataset=parameter_or_dataset.dataset,
+                    period=period,
+                    month_of_year=first_day_of_month_to_fetch.strftime("%Y%m"),
+                )
+                downloaded_file = download_file(
+                    url=url,
+                    cache_dir=self.sr.stations.settings.cache_dir,
+                    ttl=CacheExpiry.FIVE_MINUTES,
+                    client_kwargs=self.sr.settings.fsspec_client_kwargs,
+                    cache_disable=self.sr.settings.cache_disable,
+                )
+
+                if downloaded_file.status == 404:
+                    log.warning(
+                        f"File {url.rsplit('/', 1)[1]} for station {station_id} not found on server {url}. Skipping."
                     )
```

Comment:
I think this should only be an info. Warning would mean something is off but a missing file in my view is rather something "normal". What do you think?